src/annotator/plugin/bucket-bar.js

```diff
@@ -69,133 +97,181 @@ export default class BucketBar extends Delegator {
+        const index = this.tabs.indexOf(tabEl);
+        if (
+          index > 0 &&
+          this.buckets[index] &&
+          !this.isNavigationBucket(index)
+        ) {
+          focusHighlights(this.buckets[index], true);
+        }
+      });
+
+      tabEl.addEventListener('mouseout', () => {
+        const index = this.tabs.indexOf(tabEl);
+        if (index > 0 && this.buckets[index]) {
+          focusHighlights(this.buckets[index], false);
+        }
+      });
 
-        .on('mouseout', event => {
-          const bucket = this.tabs.index(event.currentTarget);
-          this.buckets[bucket].forEach(anchor =>
-            $(anchor.highlights).removeClass('hypothesis-highlight-focused')
+      // Select the annotations (in the sidebar)
+      // that have anchors within the clicked bucket
+      tabEl.addEventListener('click', event => {
+        event.stopPropagation();
+        const index = this.tabs.indexOf(tabEl);
+        const bucket = this.buckets[index];
+        if (!bucket) {
+          return;
+        }
+        if (this.isLower(index)) {
+          scrollToClosest(bucket.anchors, 'down');
+        } else if (this.isUpper(index)) {
+          scrollToClosest(bucket.anchors, 'up');
+        } else {
+          const annotations = bucket.anchors.map(anchor => anchor.annotation);
+          this.annotator.selectAnnotations(
+            annotations,
+            event.ctrlKey || event.metaKey
           );
-        })
-        .on('click', event => {
-          const bucket = this.tabs.index(event.currentTarget);
-          event.stopPropagation();
-
-          // If it's the upper tab, scroll to next anchor above
-          if (this.isUpper(bucket)) {
-            scrollToClosest(this.buckets[bucket], 'up');
-            // If it's the lower tab, scroll to next anchor below
-          } else if (this.isLower(bucket)) {
-            scrollToClosest(this.buckets[bucket], 'down');
-          } else {
-            const annotations = this.buckets[bucket].map(
-              anchor => anchor.annotation
-            );
-            this.annotator.selectAnnotations(
-              annotations,
-              event.ctrlKey || event.metaKey
-            );
-          }
-        });
+        }
+      });
+
+      this.element.append(tabEl);
     });
 
     this._buildTabs();
   }
 
   _buildTabs() {
-    this.tabs.each((index, el) => {
-      let bucketSize;
-      el = $(el);
-      const bucket = this.buckets[index];
-      const bucketLength = bucket?.length;
-
-      const title = (() => {
-        if (bucketLength !== 1) {
-          return `Show ${bucketLength} annotations`;
-        } else if (bucketLength > 0) {
-          return 'Show one annotation';
+    this.tabs.forEach((tabEl, index) => {
+      const anchorCount = this.buckets[index].anchors.length;
+      // Positioning logic currently _relies_ on their being interstitial
+      // buckets that have no anchors but do have positions. Positioning
```

Comment:
It took me a while to understand why these empty buckets exist and what their position values mean. My understanding from a little debugging is that after `update()` runs, the `this.buckets` array looks like this:

```
1. [empty bucket]
2. [bucket with above-screen anchors]
3. [empty bucket]

4. [first bucket with onscreen anchors]
5. [empty bucket]

6. [second bucket with on-screen anchors]
7. [empty bucket]

8. [empty bucket]
9. [bucket with below-screen anchors]
10. [empty bucket]
```

For the buckets that contain on-screen anchors, the position value is the top offset of the associated highlights. Each of these is followed by a bucket that is empty where the position value is the bottom offset of the associated highlights in the previous bucket. We want the visible bucket indicator to point to the middle of the highlight, hence the positions of buckets with anchors needs to be averaged with the subsequent sibling bucket.

Have I understood this correctly?

The next follow-up question I have is why the first and last hidden buckets exist.

Reply:
> It took me a while to understand why these empty buckets exist and what their position values mean

Welcome to the club.

> followed by a bucket that is empty where the position value is the bottom offset of the associated highlights in the previous bucket

Thank you for wording this this way! It was sort an a-ha moment for me. The `bottom offset` bit. Thanks. I've been staring at this logic for quite some time now but fresh eyes help make sense of it.